cmd/security-agent/app/compliance.go

```diff
@@ -64,16 +77,118 @@ func eventRun(cmd *cobra.Command, args []string) error {
 		return fmt.Errorf("failed to set up compliance log reporter: %w", err)
 	}
 
-	eventArgs.event.Data = event.Data{}
+	eventData := event.Data{}
 	for _, d := range eventArgs.data {
 		kv := strings.SplitN(d, ":", 2)
 		if len(kv) != 2 {
 			continue
 		}
-		eventArgs.event.Data[kv[0]] = kv[1]
+		eventData[kv[0]] = kv[1]
 	}
+	eventArgs.event.Data = eventData
 
 	reporter.Report(&eventArgs.event)
 
 	return nil
 }
+
+func newComplianceReporter(stopper restart.Stopper, sourceName, sourceType string) (event.Reporter, error) {
+	httpConnectivity := config.HTTPConnectivityFailure
+	if endpoints, err := config.BuildHTTPEndpoints(); err == nil {
+		httpConnectivity = http.CheckConnectivity(endpoints.Main)
+	}
+
+	endpoints, err := config.BuildEndpoints(httpConnectivity)
```

Comment:
You're right.
1. If I remember correctly, this design choice was done because logs collection may be disabled on the agent. Any other idea @xornivore ?
2. I removed the DDClient to be more consistent
3. I updated the PR so that config.Endpoints and client.DestinationsContext are shared.

I also added a comment in config_template.yaml to explain to explain that configuration settings required to connect to DD are fetch from the `logs_config` section.

Reply:
Yes, so logs collection being disabled is one concern. Another consideration is that the logs intake will eventually be different for regular logs and the security agent events.

Reply:
Ok, thanks for the context